src/students/students.service.spec.ts

```diff
@@ -1,12 +1,50 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { StudentsService } from './students.service';
+import { DBService } from '../database/db.service';
+import { CreateStudentDto } from './dto/create-student.dto';
+import { UpdateStudentDto } from './dto/update-student.dto';
 
 describe('StudentsService', () => {
   let service: StudentsService;
 
+  let mockStudentDB = [];
+  const mockDBService = {
+    student: {},
+  };
+  mockDBService.student = {
+    findMany: jest.fn(() => mockStudentDB),
+    findFirst: jest.fn(({ where: { id } }) =>
+      mockStudentDB.find((student) => student.id === id),
+    ),
+    create: jest.fn(({ data: student }) => {
+      const newStudent = {
+        id: mockStudentDB.length + 1,
+        ...student,
+      };
+      mockStudentDB.push(newStudent);
+      return newStudent;
+    }),
+    update: jest.fn(({ where: { id }, data: student }) => {
+      const updatedStudent = {
+        id: id,
+        ...student,
+      };
+      const index = mockStudentDB.findIndex((s) => s.id === id);
+      mockStudentDB[index] = updatedStudent;
+      return updatedStudent;
+    }),
+    delete: jest.fn(
+      ({ where: { id } }) =>
+        (mockStudentDB = mockStudentDB.filter((s) => s.id !== id)),
```

Comment:
I was expecting to have two individual branches . one will cover the unit test of Controller only and another will cover the Service Unit test only

Reply:
well I pushed separate branches but due to unstable changes in dev branches, fixing repetitive merge conflicts from separate branches might result in inconsistency

Reply:
So I tied all the changes to one branch for better handling and comfort